tests/Aspire.Hosting.Tests/Dcp/MockKubernetesService.cs

```diff
@@ -11,7 +11,20 @@ internal sealed class MockKubernetesService : IKubernetesService
 {
     internal sealed record DeletedResource(Type Type, object Value);
 
-    public List<object> CreatedResources { get; } = [];
+    public List<CustomResource> CreatedResources { get; } = [];
+
+    public Task<T> GetAsync<T>(string name, string? namespaceParameter = null, CancellationToken _ = default) where T : CustomResource
+    {
+        var res = CreatedResources.OfType<T>().FirstOrDefault(r =>
+            r.Metadata.Name == name &&
+            string.Equals(r.Metadata.NamespaceProperty ?? string.Empty, namespaceParameter ?? string.Empty)
+        );
+        if (res == null)
+        {
+            throw new ArgumentException($"Resource '{name}' not found");
```

Comment:
Use the overload the specifies the param name too?

Reply:
Actually, what is missing here is the reference to namespace, if provided. So I do not think I will use the overload with the param name for creating the Exception (there might be more than one parameter involved), but I need to include the namespace--thanks!

Reply:
Done. Just to clarify: the `somenamespace/someobject` is a Kubernetes convention for presenting object names; and if the object is not namespaced, then that becomes `/someobject`